jedi/inference/names.py

```diff
@@ -24,8 +25,8 @@ def _merge_name_docs(names):
 
 
 class AbstractNameDefinition(object):
-    start_pos = None
-    string_name = None
+    start_pos: Optional[Tuple[int, int]] = None
+    string_name: Optional[str] = None
```

Comment:
I've assumed that these (and elsewhere) are legitimately `Optional`. If that's not the case and these are `None` for lack of a better placeholder value (and should in fact _always_ be provided by a child class), then the way to type that would be to use the non-`Optional` type and remove the assignment:
```python
    start_pos: Tuple[int, int]
    string_name: str
```

Reply:
0571e126